pkgs/development/libraries/librsvg/default.nix

```diff
@@ -114,10 +126,14 @@ stdenv.mkDerivation rec {
 
     # 'error: linker `cc` not found' when cross-compiling
     export RUSTFLAGS="-Clinker=$CC"
+  '' + lib.optionalString ((stdenv.buildPlatform != stdenv.hostPlatform) && (stdenv.hostPlatform.emulatorAvailable buildPackages)) ''
+    # the replacement is the native conditional
+    substituteInPlace gdk-pixbuf-loader/Makefile \
+      --replace 'RUN_QUERY_LOADER_TEST = false' 'RUN_QUERY_LOADER_TEST = test -z "$(DESTDIR)"' \
   '';
 
   # Not generated when cross compiling.
-  postInstall = lib.optionalString (stdenv.hostPlatform == stdenv.buildPlatform) ''
+  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
```

Comment:
With this PR, I also need to add this diff for librsvg to build with cross again:

```diff
diff --git a/pkgs/development/libraries/librsvg/default.nix b/pkgs/development/libraries/librsvg/default.nix
index 9974826573a..87ddcd28403 100644
--- a/pkgs/development/libraries/librsvg/default.nix
+++ b/pkgs/development/libraries/librsvg/default.nix
@@ -147,11 +147,11 @@ stdenv.mkDerivation (finalAttrs: {
     mv $GDK_PIXBUF/loaders.cache.tmp $GDK_PIXBUF/loaders.cache
 
     mkdir -p "$out/share/bash-completion/completions/"
-    $out/bin/rsvg-convert --completion bash > "$out/share/bash-completion/completions/rsvg-convert"
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert --completion bash > "$out/share/bash-completion/completions/rsvg-convert"
     mkdir -p "$out/share/zsh/site-functions/"
-    $out/bin/rsvg-convert --completion zsh > "$out/share/zsh/site-functions/_rsvg-convert"
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert --completion zsh > "$out/share/zsh/site-functions/_rsvg-convert"
     mkdir -p "$out/share/fish/vendor_completions.d/"
-    $out/bin/rsvg-convert --completion fish > "$out/share/fish/vendor_completions.d/rsvg-convert.fish"
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert --completion fish > "$out/share/fish/vendor_completions.d/rsvg-convert.fish"
   '';
 
   postFixup = lib.optionalString withIntrospection ''
```

Is this the correct approach for fixing that?

If so, I'll open a PR to fix librsvg cross (I'll probably use a `let rsvg-convert = "${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert" in ...` to shorten it a little, though)

Reply:
We already have that on staging: https://github.com/NixOS/nixpkgs/pull/244602

Reply:
Ah, thanks! I did a quick search but missed that. Sorry for the noise 😅